Formula/f/flint.rb

```diff
@@ -21,25 +21,39 @@ class Flint < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "796f363a97eb42772140b94d31a9eaccfb2acf445175639aba7869250b293942"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
   depends_on "gmp"
   depends_on "mpfr"
-  depends_on "ntl"
-
   uses_from_macos "m4" => :build
 
   def install
+    # to build against NTL
     ENV.cxx11
+
+    system "./bootstrap.sh" if build.head?
+
     args = %W[
       --with-gmp=#{Formula["gmp"].prefix}
       --with-mpfr=#{Formula["mpfr"].prefix}
-      --with-ntl=#{Formula["ntl"].prefix}
     ]
-    if build.bottle?
-      args << "ax_cv_check_cxxflags___march_native=no"
-      args << "ax_cv_check_cflags___march_native=no"
+
+    if Hardware::CPU.intel?
+      # enable/disable avx{2,512}
+      # Because flint doesn't use CPUID at runtime
+      # we cannot rely on -march options
+      if build.bottle?
+        # prevent avx{2,512} in case we are building on a machine that supports it
+        args << "--enable-arch=#{Hardware.oldest_cpu}"
+      elsif Hardware::CPU.avx2?
+        # TODO: enable avx512 support
+        args << "--enable-avx2"
```

Comment:
Flint doesn't use BLAS yet, right? If it did, Apple's Accelerate framework has SIMD vector operations that replace avx: https://developer.apple.com/documentation/accelerate/simd

Reply:
We do use it in multiplication for the `nmod_mat` and `fmpz_mat` modules.

Reply:
BLAS is optional; we could compile against it and add it as a dependency.
@albinahlback and @fredrik-johansson, is this something that we want?